### DISCUSSION TITLE
Hydration is not happening in next js with zustand

### DISCUSSION THREAD

**Discussion by saishankarmanugula** [pmndrs/zustand#2690] [Q&A]
We are having a nextjs(14) app router application with zustand state management. During SSR, we are prefilling the form data for the page. On client side we have some events based on change of one of the dropdown field. But user is changing the dropdown value before the hydration(mostly happening in slow networks) which doesnt trigger any event handlers as hydration didnt complete. We are expecting that after hydration is done, page should again be in the state that it is rendered during SSR changing dropdown value to default irrespective of user changes before hydration. But it is not happening.

Its working fine if we set the value in useEffect but we have many components where this need to be done.

Am I missing anything.

Thanks

**Comment by dbritto-dev** [comment]
@saishankarmanugula would you mind sharing us a minimal repro on stackblitz?

**Reply by saishankarmanugula** [reply]
@dbritto-dev This is one example(https://stackblitz.com/edit/stackblitz-starters-5whelt?file=app%2Fcomponents%2Flocation-type.jsx) but not able to mimic the slow network behaviour. Issue I am mentioning is , User changed the location value to Business before hydration. This doesn't show the input field as the event listeners are not attached before hydration. I am expecting that location dropdown value should default to 'Residence'(similar to SSR rendered html) once hydration is done.

**Reply by dbritto-dev** [reply]
@saishankarmanugula as far as I can see you are using global stores, that's wrong. I encourage you to read our nextjs guide.

I won't be able to help you if you don't follow our recommendations about nextjs setup.

**Reply by saishankarmanugula** [reply]
@dbritto-dev We are actually using per request store only.  Changed the stackbliz accordingly(https://stackblitz.com/edit/stackblitz-starters-5whelt?file=app%2Fcomponents%2Fwrapper.jsx) Please check

**Reply by dbritto-dev** [reply]
@saishankarmanugula the new version looks good to me, did you still facing the same issue?

**Reply by saishankarmanugula** [reply]
yaeh @dbritto-dev . But, I was not able to reproduce the exact sceanrio where there is some time gap between the initial html response and the hydration as the size of the build is less in stackblitz. I can observe it when there is some delay for hydration to happen.

**Reply by dbritto-dev** [reply]
@saishankarmanugula you can use network tab -> 
<img width="552" alt="image" src="https://github.com/user-attachments/assets/630b9bc9-072f-4f3c-8bda-68bf2fc48dad">

**Reply by saishankarmanugula** [reply]
@dbritto-dev I have deployed the app in vercel - https://stackblitz-starters-5whelt.vercel.app/. We are able to reproduce using this. Can you quickly change the location to business on initial page load before hydration. Observe that location type didn't get updated to 'Residence' even after hydration.

**Reply by dbritto-dev** [reply]
<img width="1088" alt="image" src="https://github.com/user-attachments/assets/3ed100de-6cf0-4833-99bd-459d28a4a179">

**Reply by saishankarmanugula** [reply]
@dbritto-dev Its working fine for me. Can you share the console logs or try in different browser/incognito once.

![image](https://github.com/user-attachments/assets/9699fb68-ca50-4fd8-823a-b2efa7eef529)
